lib/scholar/cross_decomposition/pls_svd.ex

```diff
@@ -0,0 +1,356 @@
+    {x, y, x_mean, y_mean, x_std, y_std} = center_scale_x_y(x, y, opts)
+
+    c =
+      Nx.transpose(x)
+      |> Nx.dot(y)
+
+    {u, _s, vt} = Nx.LinAlg.svd(c, full_matrices?: false)
+    u = Nx.slice_along_axis(u, 0, num_components, axis: 1)
+    vt = Nx.slice_along_axis(vt, 0, num_components, axis: 0)
+    {u, vt} = Scholar.Decomposition.Utils.flip_svd(u, vt)
+    v = Nx.transpose(vt)
+
+    x_weights = u
+    y_weights = v
+
+    %__MODULE__{
+      x_mean: x_mean,
+      y_mean: y_mean,
+      x_std: x_std,
+      y_std: y_std,
+      x_weights: x_weights,
+      y_weights: y_weights
+    }
+  end
+
+  @doc """
+    Apply the dimensionality reduction.
+    Takes as arguments: 
+
+    * fitted estimator struct which is return value of `fit/3` function from this module
+
+    * `x` - training samples, `{num_samples, num_features}` shaped tensor
+    
+    * `y` - targets, `{num_samples, num_targets}` shaped `y` tensor
+
+  ## Options
+
+  #{NimbleOptions.docs(@opts_schema)}
+
+  ## Return Values
+    
+    Returns tuple with transformed data `{x_transformed, y_transformed}` where:
+
+    * `x_transformed` is `{num_samples, num_features}` shaped tensor.
+    
+    * `y_transformed` is `{num_samples, num_features}` shaped tensor.
+
+  ## Examples
+
+      iex> x = Nx.tensor([[0.0, 0.0, 1.0],
+      ...>                [1.0, 0.0, 0.0],
+      ...>                [2.0, 2.0, 2.0],
+      ...>                [2.0, 5.0, 4.0]])
+      iex> y = Nx.tensor([[0.1, -0.2],
+      ...>                [0.9, 1.1],
+      ...>                [6.2, 5.9],
+      ...>                [11.9, 12.3]])
+      iex> model = Scholar.CrossDecomposition.PLSSVD.fit(x, y)
+      iex> {x, y} = Scholar.CrossDecomposition.PLSSVD.transform(model, x, y)
+      iex> x
+      #Nx.Tensor<
+        f32[4][2]
+        [
+          [-1.397004246711731, -0.10283949971199036],
+          [-1.1967883110046387, 0.17159013450145721],
+          [0.5603229403495789, -0.10849219560623169],
+          [2.0334696769714355, 0.039741579443216324]
+        ]
+      >
+      iex> y
+      #Nx.Tensor<
+        f32[4][2]
+        [
+          [-1.2260178327560425, -0.019306711852550507],
+          [-0.9602956175804138, 0.04015407711267471],
+          [0.3249155580997467, -0.04311027377843857],
+          [1.8613981008529663, 0.022262824699282646]
+        ]
+      >
+
+  """
+  deftransform transform(model, x, y, opts \\ []) do
+    transform_n(model, x, y, NimbleOptions.validate!(opts, @opts_schema))
+  end
+
+  defnp transform_n(
+          %__MODULE__{
+            x_mean: x_mean,
+            y_mean: y_mean,
+            x_std: x_std,
+            y_std: y_std,
+            x_weights: x_weights,
+            y_weights: y_weights
+          } = _model,
+          x,
+          y,
+          opts
+        ) do
+    {x, y} = check_x_y(x, y, opts)
+
+    xr = (x - x_mean) / x_std
+    x_scores = Nx.dot(xr, x_weights)
+
+    yr = (y - y_mean) / y_std
+    y_scores = Nx.dot(yr, y_weights)
```

Comment:
```suggestion
    y_scores = Nx.dot(yr, [1], y_weights, [1])
```

if you remove the Nx.transpose from the definition in `y_weights` you can use this suggestion to achieve the same result

Reply:
Is it more efficient to call `Nx.dot` like you suggest? I’d like to understand why you're insisting on doing it this way.

Reply:
It is because you avoid the explicit transpose when producing the tensor.

Nx.dot(a, Nx.transpose(b))

is equivalent to

Nx.dot(a, [1], b, [1])